src/commands.ts

```diff
@@ -517,10 +517,19 @@ async function editCMD(shell:Shell, args: string[]) {
 }
 
 async function hostsCMD(shell: Shell) {
-    let res = ""
-    terminal7.gates.forEach(gate => {
-        res += `\x1B[1m${gate.name}:\x1B[0m ${gate.addr} ${gate.fp || ""}\n`
-    })
+    const attrs = terminal7.gates.map(
+        g => [g.name, `${g.username || "<unset>"}@${g.addr}`, g.fp?.slice(0, 16) || ""])
```

Comment:
replace unset with TBD and handle the case when there's no `addr` - when the gate is pure PeerBook. In this case, the column should be left empty.

Reply:
Regarding the fingerprint, I liked the way openai cuts the keys: the print the first 4 digit, "...", and the last 4 digits.